x-pack/plugins/task_manager/server/MONITORING.md

```diff
@@ -0,0 +1,242 @@
+# Task Manager Monitoring
+
+Task Manager has an internal monitoring mechanism in which keeps track of a variety of metrics which are exposed via a `health` api endpoint and Kibana Server Log debug messaging.
+
+## Exposed Metrics
+There are three different sections to the stats returned by the `health` api.
+- `configuration`: Summarizes Task Manager's current configuration.
+- `workload`: Summarizes the workload in the current deployment.
+- `runtime`: Tracks Task Manager's performance.
+
+### Configuring the Stats
+There are three new configurations:
+
+- `xpack.task_manager.monitored_stats_required_freshness` - The _required freshness_ of critical "Hot" stats, which means that if key stats (last polling cycle time, for example) haven't been refreshed within the specified duration, the `_health` endpoint and service will report an `Error` status. By default this is inferred from the configured `poll_interval` and is set to `poll_interval` plus a `1s` buffer.
+- `xpack.task_manager.monitored_aggregated_stats_refresh_rate` - Dictates how often we refresh the "Cold" metrics.  These metrics require an aggregation against Elasticsearch and add load to the system, hence we want to limit how often we execute these. We also inffer the _required freshness_ of these "Cold" metrics from this configuration, which means that if these stats have not been updated within the required duration then the `_health` endpoint and service will report an `Error` status. This covers the entire `workload` section of the stats. By default this is configured to `60s`, and as a result the _required freshness_ defaults to `61s` (refresh plus a `1s` buffer).
```

Comment:
So this is the rate at which we do the queries, and there is also a required freshness for this?  Isn't that freshness then just the latency to make the request and get the response?  Also seems like we'd have some kind of timeout on the call, and retry, and log a warning when we had to retry.

Reply:
That's designed to _catch_ the case where the workload hasn't been updates because something in the mechanism broke.
When building a monitoring endpoint you also need to take into account that you want a warning if the monitoring itself has broken... this ensures that if, for some reason, the code that fetches the workload stalls, you don't just have a green light on your monitoring because it isn't refreshing and the last value was OK.

Reply:
This of this as two separate components - one that keeps the monitored stats up to date and one that looks at these monitored stats and assesses the health based on them.